modules/guardduty/common/README.md

```diff
@@ -0,0 +1,158 @@
+# Component: `guardduty/common`
+
+This component is responsible for configuring GuardDuty and it should be used in tandem with the [guardduty/root](../guardduty/root) component.
+
+AWS GuardDuty is a managed threat detection service. It is designed to help protect AWS accounts and workloads by continuously monitoring for malicious activities and unauthorized behaviors. GuardDuty analyzes various data sources within your AWS environment, such as AWS CloudTrail logs, VPC Flow Logs, and DNS logs, to detect potential security threats.
+
+Key features and components of AWS GuardDuty include:
+
+- Threat detection: GuardDuty employs machine learning algorithms, anomaly detection, and integrated threat intelligence to identify suspicious activities, unauthorized access attempts, and potential security threats. It analyzes event logs and network traffic data to detect patterns, anomalies, and known attack techniques.
+
+- Threat intelligence: GuardDuty leverages threat intelligence feeds from AWS, trusted partners, and the global community to enhance its detection capabilities. It uses this intelligence to identify known malicious IP addresses, domains, and other indicators of compromise.
+
+- Real-time alerts: When GuardDuty identifies a potential security issue, it generates real-time alerts that can be delivered through AWS CloudWatch Events. These alerts can be integrated with other AWS services like Amazon SNS or AWS Lambda for immediate action or custom response workflows.
+
+- Multi-account support: GuardDuty can be enabled across multiple AWS accounts, allowing centralized management and monitoring of security across an entire organization's AWS infrastructure. This helps to maintain consistent security policies and practices.
+
+- Automated remediation: GuardDuty integrates with other AWS services, such as AWS Macie, AWS Security Hub, and AWS Systems Manager, to facilitate automated threat response and remediation actions. This helps to minimize the impact of security incidents and reduces the need for manual intervention.
+
+- Security findings and reports: GuardDuty provides detailed security findings and reports that include information about detected threats, affected AWS resources, and recommended remediation actions. These findings can be accessed through the AWS Management Console or retrieved via APIs for further analysis and reporting.
+
+GuardDuty offers a scalable and flexible approach to threat detection within AWS environments, providing organizations with an additional layer of security to proactively identify and respond to potential security risks.
+
+## Usage
+
+**Stack Level**: Regional
+
+The example snippet below shows how to use this component:
+
+```yaml
+components:
+  terraform:
+    guardduty/common:
+      metadata:
+        component: guardduty/common
+      vars:
+        enabled: true
+        account_map_tenant: core
+        central_resource_collector_account: core-security
```

Comment:
should be `core-security` since we are doing lookup in `full_account_map` which looks like this

```
+ full_account_map               = {
  + core-artifacts   = "111111111111"
  + core-audit       = "222222222222"
  + core-auto        = "333333333333"
  ...
  + plat-dev         = "444444444444"
  + plat-prod        = "555555555555"
  + plat-sandbox     = "666666666666"
  ...
}
```